iroha-lib/model/Tx.hpp

```diff
@@ -0,0 +1,73 @@
+#ifndef TX_HPP
+#define TX_HPP
+
+#include "LibsAndClassDeclarations.h"
+
+#include "transaction.pb.h"
+#include <boost/bimap.hpp>
+#include "crypto/keypair.hpp"
+#include "generators/CommandGenerator.hpp"
+
+
+namespace iroha_lib
+{
+
+class Tx
+{
+    iroha::keypair_t keypair_;
+    iroha::protocol::Transaction protobuf_transaction_;
+    CommandGenerator cmd_generator_;
+
+public:
+    Tx(const std::string& account_id,
+       const iroha::keypair_t& keypair,
+       uint64_t created_time = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count(),
+       uint32_t quorum = 1);
+
+    void addCommand(const iroha::protocol::Command& command);
+
+    Tx& addAssetQuantity(const std::string& account_id, const std::string& role_name);
+    Tx& addPeer(const std::string& address,
+                const std::string& pubkey,
+                const std::optional<std::string>& tls_certificate,
+                bool syncing_peer);
```

Comment:
I suggest default values - it would be easier to use:
```suggestion
                const std::optional<std::string>& tls_certificate = {},
                bool syncing_peer = false);
```

Reply:
Added.